### DISCUSSION TITLE
Chromecast Support - Beta  Testing

### DISCUSSION THREAD

**Discussion by PRProd** [PRProd/HA-Firemote#113] [Announcements]
BETA RELEASE - v2.3.0-b1

## What's New:
 - Official support for Chromecast HD

## Known Issues:
 - Navigation buttons will not work from inside the Netflix app if you've optionally paired your Firemote with the Android TV Remote integration.  [Issue can be followed here](https://github.com/home-assistant/core/issues/92659)

![chromecastRemote](https://github.com/PRProd/HA-Firemote/assets/25486599/d9d1ff2b-5f2d-4752-9a4f-ced48a38c921)

## TESTERS NEEDED!
Please join us!  Leave your experiences in a note below, positive or negative.
@huizebruin @danny77 @jrspowers and I will all appreciate your help!

**Comment by PRProd** [comment]
I'm curious if anybody noticed the missing volume buttons, and/or if I should try to add them on the side like the real remote.  My concern with that approach is that they would be very small and maybe a little hard to tap.

**Reply by danny77** [reply]
Indeed, on the real remote they on the side, i think it will be easier to put them on top on the bottom.

**Reply by xavc12011** [reply]
Can volumen buttons been added  down app buttons or maybe downside power button. It’s very important to me get then on hand
Nice work keep rocking!

**Reply by PRProd** [reply]
I'm working on this right now.  :smile:

**Reply by PRProd** [reply]
Something like this?
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/5cea7342-427c-42e9-8104-58466508b3b3)

**Reply by danny77** [reply]
This is looking very nice. ;-)

**Reply by xavc12011** [reply]
As seen on my dreams

**Reply by PRProd** [reply]
Release [v2.3.3](https://github.com/PRProd/HA-Firemote/releases/tag/v2.3.3) is now available for download, and it includes the volume buttons for the Chromecast remote!

**Reply by danny77** [reply]
Cool thank you very much.

**Reply by danny77** [reply]
The volume buttons doesn't seem to work, maybe it is possible that i´m doing something wrong ? Is there a way to test this ?

**Reply by xavc12011** [reply]
Mine the same no volume response from button control

**Reply by PRProd** [reply]
@danny77 & @xavc12011 - Thanks for letting me know.  Your note made me take a closer look at the [FAQ section](https://github.com/PRProd/HA-Firemote/tree/main#faq) in the [read me file](https://github.com/PRProd/HA-Firemote/blob/main/README.md) and I found that it needed some adjustments for clarity.

If you take a look at the updated [Why won't the volume, mute, and/or power buttons work from my Firemote card?](https://github.com/PRProd/HA-Firemote/tree/main#why-wont-the-volume-mute-andor-power-buttons-work-from-my-firemote-card) section, you will see that the most likely reason is that you have programmed your Chromecast remote control to use the IR emitter  (infrared emitter - located on the front edge of your physical remote, it looks like a black circle) to communicate directly with your TV or sound system.  You can prove this theory by simply placing your hand over the ir emitter of your physical remote, and then attempt to press the volume buttons.  If the volume buttons stop working on your physical remote when you cover up the emitter, then that's exactly what's happening.

To see or adjust the settings of your remote control, you can follow these steps.

From the home screen, navigate over to the gear icon
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/f84ad572-8543-40e4-933c-55772453db2a)

When you do, you'll see another menu pop over the top of it.  Select the gear icon here.  This will open the settings menu.
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/607e97c6-13b6-4df4-9aa6-10534c0c6eb7)

In the settings menu, scroll down until you find the option "Remotes & Accessories", and select it
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/003a0cdd-a9e7-4232-8d4c-3db9a0bbabb9)

Next, select "Set up remote buttons"
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/07d7b606-d2bd-4c31-8150-41bfedc6bd8a)

In the image above, you can see that I have my remote buttons all set to [CEC control](https://www.howtogeek.com/207186/how-to-enable-hdmi-cec-on-your-tv-and-why-you-should/).  In this mode, the Firemote volume, power, and input buttons work exactly the same as my physical remote does.

I hope this helps.

**Reply by PRProd** [reply]
I've also added these instructions to a new wiki help article called [Using CEC Controls](https://github.com/PRProd/HA-Firemote/wiki/Using-CEC-Controls).  Please let me know if you find these steps to be helpful or if they could use some clarification.

**Reply by danny77** [reply]
My settings were set to IR/LGtv so the Cromcast and tv is reacting on the remote, for the powerbutton option CEC and Cromecast settings works but only my Chromcast will turn off then. For the volume buttons only the setting Chromecast is working when i put it on CEC i get no resopnse. The input button is not working on all settings and only works on IR/LGtv on the real remote, im not using this button at all so for me no problem. I guess my TV is to old to understand CEC.
Thank u very much.

**Reply by PRProd** [reply]
@danny77 - Some TVs only allow CEC communication on specific HDMI ports.  Many TVs have the option to enable or disable CEC controls from inside their individual menus.  Before you give up, it might be worthwhile to look at the physical HDMI inputs to see if there is any printed labeling on them for CEC, and also you might want to just look through your TV's setup menus.

**Comment by jrspowers** [comment]
hello what strikes me is that you always need Android Debug Bridge to do the Android tv Remote work but as I understand from the other in discord group that the chromecast with google tv is not in the Android Debug Bridge so then she can use the chromecast remote not using I had just switched off my Android Debug Bridge in home assistant then go look indeed I can choose everything but it doesn't work because I need Android Debug Bridge can you change this or am I misunderstood here the config with error message

```yaml
You need to define a Fire TV, NVIDIA Shield, Xiaomi Mi, or other Android Debug Bridge entity
type: custom:firemote-card
entity: ''
device_family: chromecast
device_type: chromecast-4k
compatibility_mode: default
android_tv_remote_entity: remote.tv_jarne
```

**Reply by PRProd** [reply]
Firemote is currently dependent on the Android Debug Bridge for a few reasons:
 - For any Amazon Fire device control, ADB is the only way to communicate since the Android TV Remote integration does not work for Fire TV devices.
 -  For the rest of the Android TV products (Shield, Xiaomi, Chromecast, etc.) they DO work with the new Android TV Remote integration, but the integration does not yet have all of the capabilities that ADB provides such as:
    - Find my remote (Shield)
    - The ability to launch all of the apps that Firemote already provides (although, I haven't done the work yet... I know that some are possible, but I can't guarantee that all of them are)
    - Likely others?

So, as of right now, Firemote still requires ADB but it will use an optional Android TV Remote integration where there is a speed benefit available, such as the navigation buttons on the Shield or Xiaomi Mi.

If that's the only thing you found so far with the Beta version for the Chromecast integration, that's awesome!  Especially since it's not even a complaint about the Chromecast integration.  :smile:

**Comment by PRProd** [comment]
Do any of you have a Chromecast HD?  I assume that it will work just the same, but I haven't tried it yet.  Also, if anybody is feeling generous, it is one of the cheapest items on my [Amazon Wish List](https://www.amazon.com/hz/wishlist/ls/2UDS10RPJ6F5K?ref_=wl_share) if you wanted to send one to me.

**Reply by danny77** [reply]
I have one at home and tested it, works perfectly.

**Comment by PRProd** [comment]
Official Release [v2.3.0](https://github.com/PRProd/HA-Firemote/releases/tag/v2.3.0) has just been published, and it includes support for Chromecast! :tada: 

Feel free to continue to use this discussion to comment on anything related to this release / addition.  I will leave it open for a while.
